sphinx/ext/napoleon/docstring.py

```diff
@@ -161,7 +186,7 @@ def __init__(self, docstring: Union[str, List[str]], config: SphinxConfig = None
             lines = docstring.splitlines()
         else:
             lines = docstring
-        self._line_iter = modify_iter(lines, modifier=lambda s: s.rstrip())
```

Comment:
I agreed that no reason to evaluate the modifier lazily. So +1 for using `peek_iter` instead of `modify_iter`.